Formula/sourcery.rb

```diff
@@ -15,10 +15,8 @@ class Sourcery < Formula
   depends_on xcode: "12.0"
 
   def install
-    system "swift", "build", "--disable-sandbox", "-c", "release", "-Xswiftc",
-                             "-target", "-Xswiftc", "x86_64-apple-macosx10.11"
+    system "swift", "build", "--disable-sandbox", "-c", "release"
     bin.install ".build/release/sourcery"
-    lib.install Dir[".build/release/*.dylib"]
```

Comment:
This changes the build for everyone; including those not on ARM. Can we use your changes only on ARM? You can use something like `if Hardware::CPU.arm?` for flow control.

Reply:
Though we should probably change the target to the appropriate macOS version? Unsure.

Reply:
It gives the same warning on x64 too: `Warning: tried to install empty array to /opt/homebrew/Cellar/sourcery/1.0.2/lib`

Typically dylib copy is needed when you are also building a kernel extension and the dylib will be loaded in apps during runtime to manage the corresponding version of the kext. It's also not smart to export swift libraries without specifying `BUILD_LIBRARY_FOR_DISTRIBUTION=YES` because you will get errors if the app is built with a different swift version.

Reply:
Hmmm. Ok, I'd be happy with applying your changes unconditionally if the test could be updated to something more meaningful. i.e. something that doesn't just do `--version` or `--help` or similar, as those can still succeed even with broken installations.

This way the test block can actually identify when the install doesn't work.

Reply:
IDK how to write the tests in ruby but this works in bash inside a temp dir:
```
echo "class A {}" > a.swift
echo "{% for type in types.all %}{{ type.name }}{% endfor %}" > a.stencil
sourcery --disableCache --sources ./a.swift --templates ./a.stencil --output ./b.swift
```
Then b.swift should look like this (including newline at end of file):
```
// Generated using Sourcery 1.0.2 — https://github.com/krzysztofzablocki/Sourcery
// DO NOT EDIT


A
```
(The command running without any error should also be enough as a test)

Reply:
You should be able to write those into files with something like
```ruby
(testpath/"a.swift").write "class A {}"
(testpath/"a.stencil").write <<~EOS
  Contents of a.stencil
EOS
```
You can look at the other test blocks in the core repo for examples. `grep 'write <<'` might be a good start.